src/Microsoft.Extensions.Http/HttpMessageHandlerBuilder.cs

```diff
@@ -35,5 +37,52 @@ public abstract class HttpMessageHandlerBuilder
         /// <see cref="AdditionalHandlers"/>.
         /// </returns>
         public abstract HttpMessageHandler Build();
+
+        protected internal static HttpMessageHandler CreateHandlerPipeline(HttpMessageHandler primaryHandler, IEnumerable<DelegatingHandler> additionalHandlers)
```

Comment:
Made this protected static so you can subclass this without too much friction. It's boilerplate, and non-trival to get right yourself.